pkg/kubelet/volumes_linux.go

```diff
@@ -0,0 +1,58 @@
+// +build linux
+
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubelet
+
+import (
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/volume"
+)
+
+// manageVolumeOwnership modifies the given volume to be owned
+// by fsGroup.
+func (kl *Kubelet) manageVolumeOwnership(pod *api.Pod, volSpec *volume.Spec, builder volume.Builder, fsGroup int64) error {
+	stat := new(syscall.Stat_t)
+	err := syscall.Stat(builder.GetPath(), stat)
+	if err != nil {
+		glog.Errorf("Couldn't stat builder path for pod %s volume %s: %v", pod.UID, volSpec.Name, err)
+		return err
+	}
+
+	return filepath.Walk(builder.GetPath(), func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		err = kl.chownRunner.Chown(path, int(stat.Uid), int(fsGroup))
```

Comment:
rather than calling stat on teh dir and using that UID for everything, why not cast the FileInfo to (*syscall.Stat_t) and use the Uid there - that way you don't change the UID at all, ever.